Main/Source/game.cpp

```diff
@@ -5191,6 +5191,9 @@ festring game::GetDataDir()
 
 festring game::GetStateDir()
 {
+#ifdef FORCE_HOME_AS_STATE_DIR
+  return GetHomeDir()+"/.ivan/";
```

Comment:
Currently, non-UNIX system for IVAN is only Windows. And the executable for Windows is expected to run inside the game folder, so in that case `Dir` is an empty string meaning the current working directory, then `/.ivan/` is inappropriate.

Where is your high score file? Shouldn't it be in ~/.ivan/.ivan now?

Reply:
I'm sorry, you were right. Fixed. :neutral_face:

Reply:
GetHomeDir is also a confusing function name on unix systems. :p The files reside there also store some "states" (config, saves, debug info). Maybe we should merge GetHomeDir() and GetStateDir() because it is so easy to cheat and there is currently no safe method to share high score files and bone files among multiple users and limit file access. (Then I will propose removing `-DLOCAL_STATE_DIR="${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/ivan")`, which is the only hard-coded path for dynamic game data now. Another pull request will do.)

Reply:
@iology Thanks for merging! :tada: 
I think I agree with you. In addition, a new class of operating systems that take a new approach to managing system configuration and packages, such as NixOS, GuixSD, Fedora Silverblue, do not allow applications to write data to random paths. 
Although it was an old tradition to have a shared highscores and bones files in old roguelikes, but the new reality changes everything.
[The new pull request is here!](https://github.com/Attnam/ivan/pull/495)